### PR TITLE
Remove HandFactory and HandRepresentation

### PR DESCRIPTION
**Pull request by jselstad**
Removes HandFactory and HandRepresentation interfaces/base classes since nobody ever implemented them...

Also rename HandProxy back to HandRepresentation since Proxy is an ugly word.

### CODE REVIEW

File: Assets/LeapMotion/Scripts/HandRepresentation.cs [hunk 1]
```diff
@@ -1,42 +1,80 @@
...
+        handModels = new List<IHandModel>();
+      }
+      handModels.Add(model);
+      if (model.GetLeapHand() == null) {
+        model.SetLeapHand(MostRecentHand);
+        model.InitHand();
+        model.BeginHand();
+        model.UpdateHand();
+      }
+      else {
```
Comment (Amarcolina) [inline Assets/LeapMotion/Scripts/HandRepresentation.cs]: This else clause breaks our super strict uber-ultra magical coding conventions, it should be on the same line as the closing bracket as the starting if statement.

File: Assets/LeapMotion/Scripts/HandRepresentation.cs [hunk 2]
```diff
@@ -1,42 +1,80 @@
...
+    public void RemoveModel(IHandModel model) {
+      if (handModels != null) {
+        model.FinishHand();
+        handModels.Remove(model);
+      }
+    }
+
+    /** Calls Updates in IHandModels that are part of this HandRepresentation */
+    public void UpdateRepresentation(Hand hand)
+    {
```
Comment (Amarcolina) [inline Assets/LeapMotion/Scripts/HandRepresentation.cs]: Same thing with this bracket.

**Comment by Amarcolina** [comment]
Besides my tiny styling nitpicks it looks great to me!

**Comment by jselstad** [comment]
@nickjbenson We already merged postprocesses, this is the HandFactory/HandProxy PR

**Comment by nickjbenson** [comment]
OK. I created an issue and assigned you to it instead; I think it's worth not letting it fall through the cracks.

PR approoooved.

**Comment by git-shaun** [comment]
Getting multiple errors in Unity 5.6 when trying to open any scene:
```
Assets/LeapMotion/Scripts/DataStructures/Editor/Tests/RingBufferTest.cs(36,6): error CS0246: The type or namespace name `ExpectedException' could not be found. Are you missing an assembly reference?
```
```
Assets/LeapMotionModules/InteractionEngine/Scripts/Editor/Tests/BaseCallGuardTest.cs(57,6): error CS0246: The type or namespace name `ExpectedException' could not be found. Are you missing an assembly reference?
```

**Comment by nickjbenson** [comment]
Testing tools are broken in 5.6 beta, we don't _technically_ support 5.6b with UnityModules. (The errors, it's worth mentioning are benign from a development perspecitve; you just have to delete our testing infrastructure to get rid of the errors, everything else works fine.)

The Unity 5.6 beta has errors with UnityModules, that's totally expected currently. We should be testing this with Unity 5.5.

**Comment by git-shaun** [comment]
Works with 5.5. Merging.

**Pull request merged**